% this is a simplified version of 
% https://github.com/yihui/knitr/blob/master/inst/examples/knitr-beamer.Rnw
\documentclass{beamer}
\ifdefined\knitrout
  \renewenvironment{knitrout}{\begin{footnotesize}}{\end{footnotesize}}
\else
\fi
\begin{document}


\title{Analysis of Illumina Methylation arrays}
\author{Mark Dunning}
\institute[Bioinformatics Core] % (optional, but mostly needed)
{

  Cancer Research Uk\\
  Cambridge Research Institute\\
  Robinson Way\\
  Cambridge \\
}

% very important to use option [fragile] for frames containing code output!
\date[Short Occasion] % (optional)
{31st January 2014}

{
\usebackgroundtemplate{\includegraphics[width=\paperwidth]{images/criBackdrop.jpg}}
\begin{frame}
  \titlepage
\end{frame}
}
\begin{frame}{Outline}
  \tableofcontents
  % You might wish to add the option [pausesections]
\end{frame}



\section{Introduction}

\begin{frame}{DNA Methylation}
\begin{itemize}
\item A chemical modification of DNA structure that plays key role in regulating gene expression. Addition of \textit{methyl} group.
\item Occurs at \textit{CpG} locations in the genome; C followed by G
\item Areas with dense concentration of CpGs known as \textit{CpG islands}
\item Increased methylation known as \textit{hypermethylation}. Decreased methylation known as \textit{hypomethylation}
\item Aberrant gains and losses of methylation reported in the progression of cancer.
\end{itemize}
\end{frame}


\begin{frame}{Measuring DNA Methylation on microarrays}

\begin{itemize}
\item{Bisulphite-treat the sample to introduce mutations at unmethylated Cs}
\item{Umethylated Cs converted to U}
\item{Perform genotyping assay on \color{red}{two} \color{green}{colour} microarray.}
\item{Compare methylated and unmethylated signal obtained.}
\end{itemize}
\end{frame}


\begin{frame}{Quantifying methylation}
For each probe, we obtain measurements for the \textbf{M}ethylated and \textbf{U}nmethylated alleles. We define the methylation level of the probe, $\beta$, to be;

\begin{equation}
\beta =  \frac{M}{U + M + 100} 
\end{equation}

$\beta$ is the proportion of methylation for a given locus $0 < \beta < 1 $

\end{frame}

\begin{frame}[fragile]{Technology evolution}
Illumina only offer human methylation arrays
\begin{itemize}
\item{GoldenGate}
\begin{itemize}
\item{A cancer gene panel}
\item{1,500 locations per sample, 96 samples per plate}
\item{GPL9183 4468 Samples 27 Datasets}
\end{itemize}
\item{27k}
\begin{itemize}
\item{27,000 locations per sample, 12 samples per chip}
\item{GPL8490 13965 Samples 238 Datasets }
\end{itemize}
\item{450k}
\begin{itemize}
\item{450,000 locations per sample, 12 samples per chip}
\item{GPL13534 8793 Samples 160 Datasets}
\end{itemize}
TCGA also has over 6,000 samples {\tt http://cancergenome.nih.gov}
\end{itemize}

\end{frame}




\begin{frame}
We will concentrate on the analysis of 450k data
\begin{itemize}
\item{It is the currently available technology}
\item{It is an area of active research}
\item{Annual workshop}
\begin{itemize}
\item{http://www2.cancer.ucl.ac.uk/medicalgenomics/tmorris/450k.html}
\end{itemize}
\item{Online forum}
\begin{itemize}
\item{http://tinyurl.com/o7lqty8}
\end{itemize}
\item{At the moment, it is prefered to sequencing for methylation analysis}
\end{itemize}
Unless specified otherwise, code will be from the {\tt minfi} package
\end{frame}

\subsection{450k design}
\begin{frame}{Different types of probe}
\includegraphics[height=6cm]{images/probetypes.png}\\
Dedeurwaerder et al. Evaluation of the Infinium Methylation 450K technology. Future Medicine
\end{frame}

\begin{frame}
\begin{itemize}
\item Assigning methylation values to each loci is not a trivial task
\item For Type I design a pair of probes measure methylated and unmethylated in the \textit{either} the red or green channel
\item For Type II, a single probe measures \color{red}{methylated} in the \color{red}{red} channel and \color{green}{unmethylated} in the \color{green}{green}.
\end{itemize}
\end{frame}

\begin{frame}
\begin{itemize}
\item Type II probes can only tolerate three CpGs within the probe
\item Type I probes tolerate more, but assumes that all methylation loci have the same state. i.e. For a 'methylated' probe all CpGs in the probe assumed to be methylated.
\item Type I probes used in regions of high CpG density. e.g .CpG islands.
\item Earlier 27k technology used exclusively Type I probes.
\includegraphics[height=3cm]{images/locations.png}
Aryee et al (2014) Minfi: A flexible and comprehensive Bioconductor package for the analysis of Infinium DNA Methylation microarrays
\end{itemize}
\end{frame}

\section{Reading the data}
\begin{frame}{idat format}
450k arrays are commonly analysed from \textit{idat} files

http://f1000research.com/articles/2-264/v1

\includegraphics[height=6cm]{images/idat.png}

\end{frame}

\begin{frame}[fragile]
Each sample has a Red and Green idat file
<<echo=F>>=
suppressPackageStartupMessages(library(minfi))
baseDir <- system.file("extdata", package = "minfiData")
list.files(file.path(baseDir, "5723646052"))
@
A targets file / sample sheet is used to define the samples
<<echo=F>>=
targets <- read.450k.sheet(baseDir)
targets[1:2,]
@
\end{frame}

\begin{frame}[fragile]
First, we retrieve the red and green intensities from the idat files
<<>>=
RGset <- read.450k.exp(base = baseDir, targets = targets)

@

<<echo=FALSE>>=
grn <- assayDataElement(RGset, "Green")
head(grn)
@
\end{frame}

\section{QC}
\subsection{Sample QC}

\begin{frame}[fragile]{QC of samples}

<<fig.height=4>>=
pd <- pData(RGset)
densityPlot(RGset, sampGroups = pd$Sample_Group,main = "Beta")

@
\end{frame}

\begin{frame}[fragile]{QC of samples}

<<fig.height=4>>=
densityBeanPlot(RGset, sampGroups = pd$Sample_Group,
                sampNames = pd$Sample_Name)

@
\end{frame}

\begin{frame}[fragile]{QC of samples}

<<fig.height=4>>=
controlStripPlot(RGset, controls="BISULFITE CONVERSION II"
                 ,sampNames = pd$Sample_Name)
@
\end{frame}

\begin{frame}{minfi QA}
Aryee et al (2014) Minfi: A flexible and comprehensive Bioconductor package for the analysis of Infinium DNA Methylation microarrays
\includegraphics[height=5cm]{images/QA.png}
Sample identity can also be confirmed using 65 SNP probes and gender inferred from the data
\end{frame}

\subsection{Probe QC}

\begin{frame}
\begin{itemize}
\item{Probes that are not consistently \textit{detected} can be discarded.}
\begin{itemize}
\item e.g. $ >25\%$ with detection p-value $< 0.05$
\end{itemize}
\item{Annotation considerations}
\begin{itemize}
\item{Cross-hybridisation}
\item{Autosomal effects}
\item{Probes that include SNPs may be influenced by sample genotype}
\end{itemize}
\end{itemize}
\end{frame}

\begin{frame}[fragile]{Detection filtering}

<<>>=
detP <- detectionP(RGset)
failed <- detP>0.05
@
Fraction of failed positions per sample
<<>>=
colMeans(failed)
colMeans(failed) > 0.01
@
How many positions failed in 50\% of samples?
<<>>=-
sum(rowMeans(failed)>0.5)
@
\end{frame}

\section{Preprocessing}

\begin{frame}[fragile]{Preprocess the intensities}
Convert the red and green intensities into Methylated and Unmethylated values using a \textit{manifest} package
<<>>=
MSet.raw <- preprocessRaw(RGset)
MSet.raw
@

\end{frame}

\begin{frame}[fragile]{Background correction}
Can mimic the background correction steps used by Illumina in GenomeStudio
<<>>=
MSet.bg <- preprocessIllumina(RGset)
args(preprocessIllumina)
@

\end{frame}




\section{Normalisation}

\subsection{Type II bias}



\begin{frame}[fragile]
Distribution of probe types
<<echo=F>>=
anno <- suppressMessages(getAnnotation(MSet.raw))
table(anno$Type)
@
Differences in signal
<<echo=FALSE,fig.height=3,warning=FALSE>>=
suppressPackageStartupMessages(library(ggplot2))
Methylated <- log2(getMeth(MSet.raw)[,1])
Unmethylated <- log2(getUnmeth(MSet.raw)[,1])
Beta <- getBeta(MSet.raw)[,1]
ProbeType <- anno$Type
df <- suppressMessages(melt(data.frame(Methylated, Unmethylated,Beta,ProbeType)))

suppressWarnings(ggplot(df, aes(x=value, col=ProbeType)) + geom_density() + facet_wrap(~variable,scales="free"))
@
\end{frame}


\begin{frame}{Problem definition}
\begin{itemize}
\item{TypeI and TypeII probes are shown to have different distributions}
\item{They also target different genomic regions}
\item{Need to use separate analysis and normalisation}
\end{itemize}
Dedeurwaerder et al
\end{frame}

\begin{frame}{SWAN}
Maksimovic et al. Genome Biology 2012, 13:R44 

Observed that for specific number of CpGs within the probe, distributions are comparable
<<echo=FALSE,warning=FALSE,message=FALSE,fig.height=4>>=
typeI <- getProbeInfo(MSet.raw, type = "I")[, c("Name", "nCpG")]
    typeII <- getProbeInfo(MSet.raw, type = "II")[, c("Name", "nCpG")]
    CpG.counts <- rbind(typeI, typeII)
    CpG.counts$Name <- as.character(CpG.counts$Name)

df <- melt(data.frame(Name = rownames(anno),Methylated, Unmethylated,ProbeType))
df$NumberOfCpGs <- CpG.counts$nCpG[match(df$Name,CpG.counts[,1])]

df <- df[df$NumberOfCpGs > 0 & df$NumberOfCpGs <4,]

suppressWarnings(ggplot(df, aes(x=value, col=ProbeType)) + geom_density() + facet_wrap(NumberOfCpGs~variable,scales="free",ncol=2))

@

Hence, they apply quantile normalisation to each subset.
\end{frame}

\begin{frame}{Other methods}
\begin{itemize}
\item{BMIQ}
\begin{itemize}
\item{Beta Mixture Quantile Dilation}
\item{Does not use assumptions about biological characteristics to select subsets of the data}
\item{Implemented in wateRmelon}
\end{itemize}
\item{Peak-based correction}
\begin{itemize}
\item{Rescale TypeII probes based on TypeI assuming a bimodal shape}
\item{Implemented in IMA package}
\item{Does not work well when distribution does not exhibit well-defined peaks}
\end{itemize}
\end{itemize}
\end{frame}



%\begin{frame}[fragile]{The effect of SWAN}

%<<>>=
%Mset.swan <- preprocessSWAN(RGset)
%Methylated <- log2(getMeth(Mset.swan)[,1])
%Unmethylated <- log2(getUnmeth(Mset.swan)[,1])
%%Beta <- getBeta(Mset.swan)[,1]
%ProbeType <- anno$Type
%df <- suppressMessages(melt(data.frame(Methylated, Unmethylated,Beta,ProbeType)))

%suppressWarnings(ggplot(df, aes(x=value, col=ProbeType)) + geom_density() + facet_wrap(~variable,scales="free"))
%@
%\end{frame}



\section{Choice of methylation measure}
\begin{frame}[fragile]{Quantify Methylation}
The standard is to use $\beta$ values
<<>>=
M <- getMeth(MSet.raw)
U <- getUnmeth(MSet.raw)
beta <- getBeta(MSet.raw)
beta[1:5,1:2]
M[1:5,1:2] / (M[1:5,1:2] + U[1:5,1:2])
@

\end{frame}



\begin{frame}{To be(ta) or not to be(ta)}
\begin{itemize}
\item{beta has a more natural interpretation}
\item{log-ratios are more-ameanable for analysis}
\end{itemize}
Comparison of Beta-value and M-value methods for quantifying methylation levels by microarray analysis. Du et al. BMC Bioinformatics
\begin{equation}
LR = log_2 \frac{M}{U}
\end{equation}
Sometimes (confusingly) called \textit{M-values}

\end{frame}

\begin{frame}{Conversion}
The two measures can be converted easily
\begin{equation}
\beta = \frac{2^M}{2^M +1}
\end{equation}

\begin{equation}
M = log_2\frac{\beta}{1 - \beta}
\end{equation}

\end{frame}



\section{Batch effects}

\begin{frame}{Batch effects}
\textit{'it seems that batch effects are almost always present in large-scale Infinium data sets, and they can introduce severe bias during subsequent analysis steps if no adequate countermeasures are taken.'} Bock. Analysing and interpreting DNA methylation data. Nature Reviews Genetics

\end{frame}

\begin{frame}[fragile]{Visualisation}
Multi-dimensional scaling (\textit{MDS}) plots are useful for assesing sample relations in a similar way to PCA.
<<echo=F>>=
Mset.swan <- preprocessSWAN(RGset)
@

<<fig.height=4,echo=F>>=
mdsPlot(Mset.swan, numPositions = 1000, sampGroups = pd$Sample_Group,
        sampNames = pd$Sample_Name)
@

\end{frame}

\begin{frame}[fragile]{Check other covariates}
<<echo=FALSE,fig.height=5>>=
b <- getBeta(Mset.swan)
 o <- order(-matrixStats:::rowVars(b))[1:1000]
    d <- dist(t(b[o, ]))
    fit <- cmdscale(d)
mds <- data.frame(PC1 = fit[,1], PC2 = fit[,2], Sex = pd$sex, Name=pd$Sample_Name,Age = pd$age,Chip = as.factor(pd$Slide))
ggplot(mds, aes(x = PC1, y = PC2, col=Sex, cex=Age,pch=Chip,label=Name)) + geom_point() + geom_text(vjust=-1,hjust=1,cex=4)
@

\end{frame}

\begin{frame}{Correction - ComBAT}
\begin{itemize}
\item{The current favourite method seems to be \textit{ComBAT}; implemented in the {\tt sva} package.}
\item{Uses an \textit{empirical bayes} framework}
\item{Requires that you have an adjustment variable (e.g. processing data) that you want to correct}
\item{Also specify the variable of interest in the experiment (e.g. tumour vs normal)}
\item{\textbf{No substitute for poorly-designed experiments!}}
\end{itemize}
\end{frame}


\section{Differential methylation}

\subsection{DMPs}
\begin{frame}[fragile]
The {\tt dmpFinder} method in {\tt minfi} uses an \textit{F-test} to find differentially methylated postions between groups. 
\begin{itemize}
\item{Log-ratios are used in the analysis}
\item{The model is fitted using limma}
\item{Variance-shrinkage recommended for small sample-size}
\end{itemize}
<<echo=FALSE,warning=FALSE>>=
options(width=50)
MSet.norm <- preprocessIllumina(RGset, bg.correct = TRUE,normalize = "controls", reference = 2)
mset <- MSet.norm[1:20000,]
M <- getM(mset, type = "beta", betaThreshold = 0.001)
@

<<warning=FALSE>>=
dmp <- dmpFinder(M, pheno=pd$Sample_Group,
type="categorical",shrinkVar=T)
head(dmp)
@
\end{frame}

\begin{frame}[fragile]
<<echo=F>>=
cpgs <- rownames(dmp)[1:4]
df <- melt(b[cpgs,])
df <- merge(df, pd, by.x=2, by.y=0,all=T)
ggplot(df, aes(x = Sample_Group, y=value,fill=Sample_Group)) + geom_boxplot() + facet_wrap(~X1,scales="free")+ theme(axis.text.x=element_text(angle = -90, hjust = 0))
@


\end{frame}


\begin{frame}{'bump-hunting'}
Larger Differentially Methylated Regions can be identified
\includegraphics[height=6cm]{images/bumphunting.png}
Now also available in minfi
\end{frame}


\begin{frame}{Available software}
Processing and analysis of DNA methylation data. Wilhelm-Benartzi et al. (2013)
\includegraphics[width=12cm]{images/software.png}


\end{frame}

\begin{frame}{ChAMP}
\begin{columns}
\column{.5\textwidth}
\begin{itemize}
\item{Automated workflow}
\item{Data read using minfi from idat files}
\item{Choice of normalization}
\item{Visualisation and correction of batch effects}
\item{CNA analysis}
\end{itemize}
\column{.5\textwidth}
\includegraphics[width=5cm]{images/champ.png}
\end{columns}
\end{frame}

%\begin{frame}[fragile]{This talk was brought to you by...}

%<<sessionInfo>>=

%sessionInfo()
%@

%%\end{frame}
\end{document}